### PR TITLE
Improve loadtime by removing LazyArray (and be ready for julia 1.9)

### PR DESCRIPTION
**Pull request by Moelf**
- https://github.com/JuliaLang/julia/pull/47184

```julia
#1.8.2
time julia --startup-file=no --project=. -e 'using UnROOT; t = LazyTree(UnROOT.samplefile("tree_with_jagged_array.root"), "t1") |> show'
 30  │ [20, 21, 22, 23, 24, 25, 26, 27, 28]
 31  │ []
  ⋮  │                  ⋮
                             69 rows omitted

________________________________________________________
Executed in    8.80 secs    fish           external
   usr time    8.81 secs    0.00 micros    8.81 secs
   sys time    1.53 secs  631.00 micros    1.53 secs

> time ~/Documents/github/julia/julia --startup-file=no --project=. -e 'using UnROOT; t = LazyTree(UnROOT.samplefile("tree_with_jagged_array.root"), "t1") |> show'
...
 29  │ [20, 21, 22, 23, 24, 25, 26, 27]
 30  │ [20, 21, 22, 23, 24, 25, 26, 27, 28]
 31  │ []
  ⋮  │                  ⋮
                             69 rows omitted

________________________________________________________
Executed in    5.75 secs    fish           external
   usr time    5.82 secs  675.00 micros    5.82 secs
   sys time    1.50 secs    0.00 micros    1.50 secs
```

### CODE REVIEW

**Comment by vchuravy** [comment]
What is the timing split between `using` and doing the task?

**Comment by Moelf** [comment]
yeah @vchuravy got it, the problem is LazyArray ('s dependency)
```julia
julia> @time_imports using UnROOT
      0.9 ms  Statistics
    156.3 ms  FillArrays
      2.2 ms  StaticArraysCore
    571.9 ms  StaticArrays
   1171.1 ms  ArrayLayouts
```

**Comment by Moelf** [comment]
after yeeting the LazyArray:
```
#1.8.2
  ⋮  │                  ⋮
                             69 rows omitted

________________________________________________________
Executed in    7.19 secs    fish           external
   usr time    7.32 secs    0.00 micros    7.32 secs
   sys time    1.46 secs  515.00 micros    1.46 secs


#1.9 wit the said PR
julia> @time_imports using UnROOT
    543.1 ms  StaticArrays

 29  │ [20, 21, 22, 23, 24, 25, 26, 27]
 30  │ [20, 21, 22, 23, 24, 25, 26, 27, 28]
 31  │ []
  ⋮  │                  ⋮
                             69 rows omitted

________________________________________________________
Executed in    4.05 secs    fish           external
   usr time    4.26 secs   11.66 millis    4.25 secs
   sys time    1.43 secs    3.16 millis    1.42 secs
```

File: src/UnROOT.jl [hunk 1]
```diff
@@ -44,4 +44,12 @@ include("iteration.jl")
 include("custom.jl")
 include("displays.jl")
 
+@static if VERSION >= v"1.9"
```
Comment (giordano) [inline src/UnROOT.jl:47]: This should help for any version of julia, no?  Also, top-level `@static` does nothing: top-level expressions in a packages are evaluated at compile-time anyway.

Comment (Moelf) [reply src/UnROOT.jl:47]: this doesn't help for <1.8

**Comment by Moelf** [comment]
got even faster after https://github.com/JuliaLang/julia/pull/47184

```julia
> time ~/Documents/github/julia/julia --startup-file=no --project=. -e 'using UnROOT; t = LazyTree(UnROOT.samplefile("tree_with_jagged_array.root"), "t1") |> show'
 Row │ int32_array
     │ SubArray{Int32,
─────┼──────────────────────────────────────
 1   │ []
 2   │ [0]
                             69 rows omitted

________________________________________________________
Executed in    2.61 secs    fish           external
   usr time    2.52 secs  638.00 micros    2.52 secs
   sys time    0.09 secs    0.00 micros    0.09 secs
```

**Pull request merged**